server/src/main/java/com/cloud/network/guru/GuestNetworkGuru.java

```diff
@@ -369,7 +377,14 @@ public NicProfile allocate(final Network network, NicProfile nic, final VirtualM
                 if (isGateway) {
                     guestIp = network.getGateway();
                 } else {
-                    guestIp = _ipAddrMgr.acquireGuestIpAddress(network, nic.getRequestedIPv4());
+                    if (nic.getRequestedIPv4() == null && vm.getVirtualMachine().getType() == VirtualMachine.Type.DomainRouter) {
+                        guestIp = this.acquireGuestIpAddressForVrouterRedundant(
+                                VrouterPlacement.fromString(VrouterRedundantTiersPlacement.valueIn(network.getAccountId())),
+                                network, nic.getRequestedIPv4()
+                        );
+                    } else {
+                        guestIp = _ipAddrMgr.acquireGuestIpAddress(network, nic.getRequestedIPv4());
+                    }
```

Comment:
can you make this block a method or put it in `acquireGuestIpAddressForVrouterRedundant()` maybe even with the validation below?

Reply:
I've made a more readable/clearer version in the last commit: https://github.com/apache/cloudstack/pull/3587/commits/b5c7e5e7547cce38160b7aa773646ca64245c3fd 
Let me know if you prefer like this